### DISCUSSION TITLE
where can i find more information on these values and their meaning? tfs-z etc on the server mode web ui

### DISCUSSION THREAD

**Discussion by hiqsociety** [ggml-org/llama.cpp#3134] [Q&A]
1. where can i find more information on these values and their meaning?
         llama_sample_top_k      (ctx, &cur_p, top_k, 1); 
         llama_sample_tail_free  (ctx, &cur_p, tfs_z, 1); 
         llama_sample_typical    (ctx, &cur_p, typical_p, 1); 
         llama_sample_top_p      (ctx, &cur_p, top_p, 1); 
         llama_sample_temperature(ctx, &cur_p, temp); 

i'm surprised there's no "seed" value to add for uniqueness.

2. how do i do "brand voice" for the ai? do you guys have any samples to show professional prompts engineered stuff and output results?

**Comment by KerfuffleV2** [comment]
This is only a partial answer:

You can start here: https://github.com/ggerganov/llama.cpp/tree/master/examples/main#top-k-sampling

I also wrote some stuff for my `llm-samplers` Rust crate. You probably don't care about Rust, but there's a brief description and sometimes a link to the paper where a sampling approach got researched: https://docs.rs/llm-samplers/latest/llm_samplers/samplers/index.html

Unfortunately, documentation for samplers like Mirostat, tail-free, etc is pretty sparse. You mostly have to learn by playing around with them.

**Reply by hiqsociety** [reply]
ok thx!
